src/player.js

```diff
@@ -707,6 +846,48 @@ export default class Player extends FakeEventTarget {
   }
 
   /**
+   * add off text track if there are actual text tracks associated with media
+   * setting this track is the same as calling Player's hideTextTrack
+   * @private
+   * @returns {void}
+   */
+  _addTextTrackOffOption(): void {
+    const textTracks = this.getTracks(TrackTypes.TEXT);
+    if (textTracks && textTracks.length) {
+      this._tracks.push(new TextTrack({
+        active: false,
+        index: textTracks.length,
+        kind: "subtitles",
+        label: "Off",
+        language: "off"
+      }));
+    }
+  }
+
+  _setDefaultTracks(){
+    const activeTracks = this.getActiveTracks();
+    const playbackConfig = this._config.playback;
+
+    if (playbackConfig.textLanguage == "auto"){
+      playbackConfig.textLanguage = Locale.language;
+    }
+
+    this._setDefaultTrack(TrackTypes.TEXT, playbackConfig.textLanguage, activeTracks.text);
+    this._setDefaultTrack(TrackTypes.AUDIO, playbackConfig.audioLanguage, activeTracks.audio);
+  }
+
+  _setDefaultTrack(type: string, language: string, defaultTrack: Track){
```

Comment:
I dont think that we need to send the type of the track.
We can check inside `_setDefaultTrack ` the track's instanceof.